### ISSUE TITLE
Create github action to automate testing

### ISSUE THREAD

**Issue by Fehings** [Clayton-TV/claytontv#50]
For now at least will want the tests to run on each pull request and make sure they pass. 

Tests should be created in catalogue/tests.

**Comment by ShuvojitNandi** [comment]
Do you want the workflow to run only on ubuntu-latest?

**Comment by thatgardnerone** [comment]
> Do you want the workflow to run only on ubuntu-latest?

I don't think so, it might be better to specify a specific version, such as ubuntu:24.04, since it's the most recent LTS. (Also, that's what the web server is running at the min, so having that parity would be ideal!)

Does that seem reasonable?

**Comment by ShuvojitNandi** [comment]
Sounds good.
